packages/eas-cli/src/project/publish.ts

```diff
@@ -681,43 +681,73 @@ export async function getRuntimeVersionObjectAsync(
   projectDir: string,
   vcsClient: Client
 ): Promise<{ platform: string; runtimeVersion: string }[]> {
-  for (const platform of platforms) {
-    if (platform === 'web') {
-      continue;
-    }
-    const isPolicy = typeof (exp[platform]?.runtimeVersion ?? exp.runtimeVersion) === 'object';
-    if (isPolicy) {
-      const isManaged =
-        (await resolveWorkflowAsync(projectDir, platform as EASBuildJobPlatform, vcsClient)) ===
-        Workflow.MANAGED;
-      if (!isManaged) {
-        throw new Error(
-          `You're currently using the bare workflow, where runtime version policies are not supported. You must set your runtime version manually. For example, define your runtime version as "1.0.0", not {"policy": "appVersion"} in your app config. ${learnMore(
-            'https://docs.expo.dev/eas-update/runtime-versions'
-          )}`
-        );
-      }
-    }
-  }
-
   return await Promise.all(
-    [...new Set(platforms)].map(async platform => {
-      if (platform === 'web') {
-        return { platform: 'web', runtimeVersion: 'UNVERSIONED' };
-      }
+    platforms.map(async platform => {
       return {
         platform,
-        runtimeVersion: nullthrows(
-          await Updates.getRuntimeVersionAsync(projectDir, exp, platform),
-          `Unable to determine runtime version for ${
-            requestedPlatformDisplayNames[platform]
-          }. ${learnMore('https://docs.expo.dev/eas-update/runtime-versions/')}`
+        runtimeVersion: await getRuntimeVersionForPlatformAsync(
+          exp,
+          platform,
+          projectDir,
+          vcsClient
         ),
       };
     })
   );
 }
 
+async function getRuntimeVersionForPlatformAsync(
+  exp: ExpoConfig,
+  platform: Platform,
+  projectDir: string,
+  vcsClient: Client
+): Promise<string> {
+  if (platform === 'web') {
+    return 'UNVERSIONED';
+  }
+
+  const runtimeVersion = exp[platform]?.runtimeVersion ?? exp.runtimeVersion;
+  if (typeof runtimeVersion === 'object') {
+    const policy = runtimeVersion.policy;
+
+    if ((policy as any) === 'fingerprintNativeExperimental') {
+      // TODO(wschurman): ensure prebuild has been run
```

Comment:
this is the main concern we need to address. how do we ensure that the native directories are in a "clean" state, i.e. the same state they are in which the fingerprint is generated at build time.